src/librustc/ty/mod.rs

```diff
@@ -2305,7 +2305,7 @@ impl<'a, 'gcx, 'tcx> AdtDef {
     /// Returns an iterator over all fields contained
     /// by this ADT.
     #[inline]
-    pub fn all_fields<'s>(&'s self) -> impl Iterator<Item = &'s FieldDef> {
+    pub fn all_fields<'s>(&'s self) -> impl Iterator<Item = &'s FieldDef> + Clone {
```

Comment:
Why the added Clone bound? The only use of this method I can see here seems to just consume the iterator once.

Reply:
It's used in src/librustc_typeck/check/mod.rs as follows:

```rust
let field_infos = adt.all_fields().map(|field| {
    // ...snip...
});

let non_zst_fields = field_infos.clone().filter(|(_span, zst, _align1)| !*zst);
let non_zst_count = non_zst_fields.clone().count();
// ...snip...
```

Reply:
Ah, I see, thanks.